File: OmarPakker/Source/QL_GOLD_C-Sharp/Algebra.QL.Core/Expr/UnaryExpr.cs
using System;

namespace Algebra.QL.Core.Expr
{
    public abstract class UnaryExpr<E>
    {
		public Tuple<int, int> SourcePosition { get; set; }
        public E Expr1 { get; private set; }

        public UnaryExpr(E e1)
        {
            Expr1 = e1;
        }

		public override bool Equals(object obj)
		{
			if (!(obj is UnaryExpr<E>))
			{
				return false;
			}

			return Expr1.Equals(((UnaryExpr<E>)obj).Expr1);
		}

		public override int GetHashCode()
		{
			return Expr1.GetHashCode();
		}
    }
}

File: GeorgePachitariu/ANTLR-First/src/expr/UnaryExpr.java
package expr;


public abstract class UnaryExpr extends Expression {
	protected Expression first;
	
	public UnaryExpr(Expression fst) {
		this.first=fst;
	}
	
	@Override
	public boolean equals(Object obj) {
		if (!(obj instanceof UnaryExpr))
			return false;
		UnaryExpr unaryExpr=(UnaryExpr) obj;
		return this.first.equals(unaryExpr.first);
	}
	
	@Override
	public boolean containsTreeNode(Expression e) {
		if(e.equals(this) || this.first.containsTreeNode(e) )
			return true;
		return false;
	}
}
